src/alloc.c

```diff
@@ -327,8 +327,8 @@ void jl_lambda_info_set_ast(jl_lambda_info_t *li, jl_value_t *ast)
     jl_gc_wb(li, li->ssavaluetypes);
     int i;
     for(i=0; i < nslots; i++) {
-        jl_value_t *vi = jl_cellref(vis, i);
-        jl_sym_t *name = (jl_sym_t*)jl_cellref(vi, 0);
+        jl_value_t *vi = jl_array_any_ref(vis, i);
```

Comment:
OK, so it's the occasion to make this clearer. What's the best name for it? `jl_array_ptr_ref`?